server/consumer.go

```diff
@@ -3423,17 +3573,32 @@ func (o *consumer) streamNumPending() uint64 {
 		o.npc, o.npcm = 0, 0
 	} else if o.cfg.DeliverPolicy == DeliverLastPerSubject {
 		o.npc, o.npcm = 0, 0
-		for _, ss := range o.mset.store.SubjectsState(o.cfg.FilterSubject) {
-			if o.sseq <= ss.Last {
-				o.npc++
-				if ss.Last > o.npcm {
-					o.npcm = ss.Last
+		for _, filter := range o.subjf {
+			for _, ss := range o.mset.store.SubjectsState(filter.subject) {
+				if filter.nextSeq <= ss.Last {
+					o.npc++
+					if ss.Last > o.npcm {
+						o.npcm = ss.Last
+					}
 				}
 			}
 		}
 	} else {
-		ss := o.mset.store.FilteredState(o.sseq, o.cfg.FilterSubject)
-		o.npc, o.npcm = ss.Msgs, ss.Last
+		var npc uint64
+		for _, filter := range o.subjf {
+			seq := filter.currentSeq
+
+			// We might loose state of o.msseq, so if we do recover from o.sseq
```

Comment:
We have an o.msseq?

Reply:
missed that when refactoring `o.msseq` into `o.subjf`